src/client/Client.cc

```diff
@@ -168,6 +168,70 @@ void client_flush_set_callback(void *p, ObjectCacher::ObjectSet *oset)
   client->flush_set_callback(oset);
 }
 
+std::string trim(const std::string &input) {
+  std::string ret = input;
+  ret.erase(ret.begin(), std::find_if(ret.begin(), ret.end(), [](unsigned char ch) {
+    return !std::isspace(ch);
+  }));
+  ret.erase(std::find_if(ret.rbegin(), ret.rend(), [](unsigned char ch) {
+    return !std::isspace(ch);
+  }).base(), ret.end());
+  return ret;
+}
+
+bool isNumber(const std::string &input) {
+  return std::all_of(input.begin(), input.end(), [](char ch) {
+    return isdigit(ch);
+  });
+}
+
+void Client::set_id_map(const std::string &map_file, std::unordered_map<uid_t, uid_t> &id_map, std::unordered_map<uid_t, uid_t> &id_map_remote) {
+  std::ifstream file(map_file);
+  std::string buff{};
+  while(std::getline(file, buff)) {
+    buff = trim(buff);
+    // skip comments and empty lines
+    if (buff.empty() || buff[0] == '#') {
+      continue;
+    }
+    auto local = trim(buff.substr(0, buff.find(':')));
+    auto remote = trim(buff.substr(buff.find(':')+1));
+    if(local.empty() || remote.empty() || !isNumber(local) || !isNumber(remote)) {
+      continue;
+    }
```

Comment:
if `isNumber()` fails, we should log a message with `dout(20)` as to what failed parsing:
eg.
```
dout(20) << "local uid '" << local << "' failed to parse to a number" << dendl;
```
otherwise there's no trace of what got dropped during the file parsing

Also, instead of implementing your own `isNumber()` and then doing a `std::stroul()`, you could also use `boost::lexical_cast<>()`, which does all the necessary things for parsing and throwing an exception if anything fails. Ceph code already uses `boost:lexical_cast<>`